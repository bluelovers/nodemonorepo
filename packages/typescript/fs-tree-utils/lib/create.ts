import {WriteTree} from './types'
import FileSystemRepresentation from './classes/fs-representation'
const {File, Directory} = FileSystemRepresentation

/**
 * Create a directory tree
 * @param tree Tree structure that needs to create
 * @param container Where to place the tree
 */
export async function create (tree: WriteTree, container: string = ''): Promise<void> {
  if (typeof tree === 'string') {
    return create(new File(tree), container)
  }

  if (typeof tree === 'function') {
    return tree(container)
  }

  if (tree instanceof FileSystemRepresentation) {
    return create(x => tree.write(x), container)
  }

  return create(new Directory(tree), container)
}

export namespace create {
  export const async = create
}

export default create
