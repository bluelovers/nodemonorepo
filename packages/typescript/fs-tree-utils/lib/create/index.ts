import * as types from '../.types'

/**
 * Create a directory tree
 * @param tree Tree structure that needs to create
 * @param container Where to place the tree
 */
export async function create (
  tree: types.Tree.Write,
  container: string
): Promise<void> {
  const { FileSystemRepresentation } = types

  if (typeof tree === 'string' || tree instanceof Buffer) {
    return create(new FileSystemRepresentation.File(tree), container)
  }

  if (typeof tree === 'function') {
    return tree(container, { create })
  }

  if (tree instanceof FileSystemRepresentation) {
    return create(x => tree.write(x, { create }), container)
  }

  return create(new FileSystemRepresentation.Directory(tree), container)
}

export namespace create {
  export const async = create
}

export default create
