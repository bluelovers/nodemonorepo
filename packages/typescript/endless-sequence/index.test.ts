import create from './index'

it('matches snapshot', () => {
  const generate =
    (previous: ReadonlyArray<string>) => previous
      .map(x => previous.map(y => x + y))
      .reduce((prev, current) => [...prev, ...current])

  const getData = () => seq.__internal.data

  const collect = (count: number) => {
    const collection = Array<string>()

    for (const element of seq) {
      if (!count) break
      --count
      collection.push(element)
    }

    return collection
  }

  const seq = create(generate, 'abcdef')
  const first = getData()

  const at2 = seq.at(2)
  const second = getData()

  const at6 = seq.at(6)
  const third = getData()

  const collection20 = collect(20)
  const forth = getData()

  const collection64 = collect(64)
  const fifth = getData()

  expect([
    {data: first},
    {at2, data: second},
    {at6, data: third},
    {collection20, data: forth},
    {collection64, data: fifth}
  ]).toMatchSnapshot()
})
