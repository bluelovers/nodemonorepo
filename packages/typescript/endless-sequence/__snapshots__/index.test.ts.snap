// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  Object {
    "data": Array [
      "a",
      "b",
      "c",
      "d",
      "e",
      "f",
    ],
  },
  Object {
    "at2": "c",
    "data": Array [
      "a",
      "b",
      "c",
      "d",
      "e",
      "f",
    ],
  },
  Object {
    "at6": "aa",
    "data": Array [
      "a",
      "b",
      "c",
      "d",
      "e",
      "f",
      "aa",
      "ab",
      "ac",
      "ad",
      "ae",
      "af",
      "ba",
      "bb",
      "bc",
      "bd",
      "be",
      "bf",
      "ca",
      "cb",
      "cc",
      "cd",
      "ce",
      "cf",
      "da",
      "db",
      "dc",
      "dd",
      "de",
      "df",
      "ea",
      "eb",
      "ec",
      "ed",
      "ee",
      "ef",
      "fa",
      "fb",
      "fc",
      "fd",
      "fe",
      "ff",
    ],
  },
  Object {
    "collection20": Array [
      "a",
      "b",
      "c",
      "d",
      "e",
      "f",
      "aa",
      "ab",
      "ac",
      "ad",
      "ae",
      "af",
      "ba",
      "bb",
      "bc",
      "bd",
      "be",
      "bf",
      "ca",
      "cb",
    ],
    "data": Array [
      "a",
      "b",
      "c",
      "d",
      "e",
      "f",
      "aa",
      "ab",
      "ac",
      "ad",
      "ae",
      "af",
      "ba",
      "bb",
      "bc",
      "bd",
      "be",
      "bf",
      "ca",
      "cb",
      "cc",
      "cd",
      "ce",
      "cf",
      "da",
      "db",
      "dc",
      "dd",
      "de",
      "df",
      "ea",
      "eb",
      "ec",
      "ed",
      "ee",
      "ef",
      "fa",
      "fb",
      "fc",
      "fd",
      "fe",
      "ff",
    ],
  },
  Object {
    "collection64": Array [
      "a",
      "b",
      "c",
      "d",
      "e",
      "f",
      "aa",
      "ab",
      "ac",
      "ad",
      "ae",
      "af",
      "ba",
      "bb",
      "bc",
      "bd",
      "be",
      "bf",
      "ca",
      "cb",
      "cc",
      "cd",
      "ce",
      "cf",
      "da",
      "db",
      "dc",
      "dd",
      "de",
      "df",
      "ea",
      "eb",
      "ec",
      "ed",
      "ee",
      "ef",
      "fa",
      "fb",
      "fc",
      "fd",
      "fe",
      "ff",
      "aa",
      "ab",
      "ac",
      "ad",
      "ae",
      "af",
      "aaa",
      "aab",
      "aac",
      "aad",
      "aae",
      "aaf",
      "aba",
      "abb",
      "abc",
      "abd",
      "abe",
      "abf",
      "aca",
      "acb",
      "acc",
      "acd",
    ],
    "data": Array [
      "a",
      "b",
      "c",
      "d",
      "e",
      "f",
      "aa",
      "ab",
      "ac",
      "ad",
      "ae",
      "af",
      "ba",
      "bb",
      "bc",
      "bd",
      "be",
      "bf",
      "ca",
      "cb",
      "cc",
      "cd",
      "ce",
      "cf",
      "da",
      "db",
      "dc",
      "dd",
      "de",
      "df",
      "ea",
      "eb",
      "ec",
      "ed",
      "ee",
      "ef",
      "fa",
      "fb",
      "fc",
      "fd",
      "fe",
      "ff",
      "aa",
      "ab",
      "ac",
      "ad",
      "ae",
      "af",
      "aaa",
      "aab",
      "aac",
      "aad",
      "aae",
      "aaf",
      "aba",
      "abb",
      "abc",
      "abd",
      "abe",
      "abf",
      "aca",
      "acb",
      "acc",
      "acd",
      "ace",
    ],
  },
]
`;
